build/bin/network-validator.go

```diff
@@ -86,11 +87,29 @@ func TestEndpoints(config reachabilityConfig) {
 }
 
 func ValidateReachability(host string, port int) error {
-	endpoint := fmt.Sprintf("%s:%d", host, port)
+	var endpoint string
+
+	httpClient := http.Client{
+		Timeout: *timeout * time.Second,
+	}
+
+	switch port {
+	case 80:
+		endpoint = fmt.Sprintf("%s://%s", "http", host)
+	case 443:
+		endpoint = fmt.Sprintf("%s://%s", "https", host)
+	case 22:
+		endpoint = fmt.Sprintf("%s://%s", "ssh", host)
+	default:
+		endpoint = fmt.Sprintf("%s://%s", "http", host)
+	}
+
 	fmt.Printf("Validating %s\n", endpoint)
-	_, err := net.DialTimeout("tcp", endpoint, *timeout)
+
+	_, err := httpClient.Get(endpoint)
 	if err != nil {
-		return fmt.Errorf("Unable to reach %s within specified timeout: %s", endpoint, err)
+		return fmt.Errorf("unable to reach %s within specified timeout: %s", endpoint, err)
```

Comment:
would this require a change also in the comparison mechanism that we're performing while validation?
https://github.com/openshift/osd-network-verifier/blob/main/pkg/cloudclient/aws/private.go#L261